Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -45,122 +60,124 @@ final class ChangeStreamTest: MongoSwiftTestCase {
-            // TODO: Use MongoCollection.watch() instead `mongoc_collection_watch` of once it gets added
-            let changeStreamPtr: OpaquePointer = mongoc_collection_watch(collPtr, pipeline._bson, opts?._bson)
-            var replyPtr = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
-            defer {
-                replyPtr.deinitialize(count: 1)
-                replyPtr.deallocate()
-            }
-
-            let decoder = BSONDecoder()
-            let changeStream = try ChangeStream<ChangeStreamDocument<Document>>(stealing: changeStreamPtr,
-                                                                                client: client,
-                                                                                connection: connection,
-                                                                                session: session,
-                                                                                decoder: decoder)
-            // expect the first iteration to be nil since no changes have been made to the collection.
-            expect(changeStream.next()).to(beNil())
-
-            // test that the change stream contains a change document for the `insert` operation.
-            try coll.insertOne(["x": 1], session: session)
-            let res1 = changeStream.next()
-            expect(res1).toNot(beNil())
-            expect(res1?.operationType).to(equal(.insert))
-            expect(res1?.fullDocument?["x"]).to(bsonEqual(1))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res1?._id))
-
-            try coll.updateOne(filter: ["x": 1], update: ["$set": ["x": 2] as Document], session: session)
-            let res2 = changeStream.next()
-
-            // test that the change stream contains a change document for the `update` operation.
-            expect(res2).toNot(beNil())
-            expect(res2?.operationType).to(equal(.update))
-            expect(res2?.fullDocument?["x"]).to(bsonEqual(2))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res2?._id))
-
-            // test that the change stream contains a change document for the `find` operation.
-            try coll.findOneAndDelete(["x": 2], session: session)
-            let res3 = changeStream.next()
-            expect(res3).toNot(beNil())
-            expect(res3?.operationType).to(equal(.delete))
-
-            // test that the resumeToken is updated
-            expect(changeStream.resumeToken).to(equal(res3?._id))
+        let changeStream = try coll.watch(options: options)
+
+        // expect the change stream to contain a change document for the `insert` operation
+        try coll.insertOne(["x": 1])
+        let res1 = changeStream.next()
+        expect(res1).toNot(beNil())
+        expect(res1?.operationType).to(equal(.insert))
+        expect(res1?.fullDocument?["x"]).to(bsonEqual(1))
```

Comment:
@patrickfreed `testChangeStreamOnACollection` creates a watch on a collection using `CollectionType` as default. It looks at the fields and checks they're not nil. Is this what you meant by [your commen](https://github.com/mongodb/mongo-swift-driver/pull/301#discussion_r308414653)t? I can add more assertions for more change events.